src/connection.ts

```diff
@@ -277,11 +277,11 @@ export class Connection extends JSForceConnection {
         .on('record', rec => records.push(rec))
         .on('error', err => reject(err))
         .on('end', () => {
-          const totalSize = getNumber(query, 'totalSize') || maxFetch;
+          const totalSize = getNumber(query, 'totalSize') || 0;
           if (totalSize > records.length) {
             process.emitWarning(
               `The query result is missing ${totalSize -
-                records.length} records due to a 10,000 limit for performance reasons. Increase the number of records returned by setting the config value "maxQueryLimit" or the environment variable "SFDX_MAX_QUERY_LIMIT" to ${totalSize} or greater than 10,000.`
+                records.length} records due to a ${maxFetch} record limit for performance reasons. Increase the number of records returned by setting the config value "maxQueryLimit" or the environment variable "SFDX_MAX_QUERY_LIMIT" to ${totalSize} or greater than 10,000.`
```

Comment:
now that thin includes overridden values, let's remove "for performance reasons". Also change the 10k to maxFetch at the end of this message.